### DISCUSSION TITLE
[Cache] Tags lifetime

### DISCUSSION THREAD

**Discussion by a-menshchikov** [symfony/symfony#45507] [Q&A]
Hi, I have a question about usage of tags in the Cache component (I'm using Redis as a storage).
I see that when I add a tag to the cache item, a new tag record (hash set in Redis) with this item identifier is created (or it's added into an existing record/hash set). But when this cache item expires, the tag record (hash set element) still exists and holds a non-existent cache item id.

Question: is there any way to clear expired cache items ids from tags (without the whole tag hash set eviction)?

**Comment by ElisDN** [comment]
You can configure limits for Redis:

```
--maxmemory 32mb
--maxmemory-policy allkeys-lru
```

if you use Redis for cache only.

**Reply by a-menshchikov** [reply]
I clarified the question about the whole tag hash set eviction.

In my use case the user identifiers used as tags, and I need a way to remove exactly the hash set elements that point to unexistent cache items. Eviction of the whole tag hash set is a wrong way in my case, because it may be in use at the moment.

**Reply by Spea** [reply]
@ElisDN Thank you for the suggestion, but setting the eviction policy to `allkeys-lru` will also not work with the `RedisTagAwareAdapter` as any other policy than `noeviction` or `volatile-*` will throw [an exception](https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Cache/Adapter/RedisTagAwareAdapter.php#L87-L89).
